### ISSUE TITLE
please add three features .

### ISSUE THREAD

**Issue by drmusti** [CanisLupus/swift-selection-search#226]
please add three features .
1) add url protocol engine . example:
es:{searchTerms} target:_self . no new tab etc. icon url or local selection.

2) add local app engine .
example:
C:\Program Files (x86)\WinCatalog\WinCatalog.exe /s "{searchTerms}" target:_self . no new tab etc. icon url or local selection.

3)possible icon offline import or offline url file:///D:/Icons/google1.ico etc...

please add features. and your extention more great...
thanks
Sincerely.

reply 02: You can already use engines like es:{searchTerms} for searching with the Everything app.
" i already using but i made config "Left mouse button click=new tab..."
because youtube google etc search engines open in new tab.
but url protocols es: test etc .. must be same tab target.
sorry my bad english?
can u understand my proplem?

**Comment by CanisLupus** [comment]
Hi there! Thanks for creating the issue here.

I think I understand at least part of it, but you duplicated some things like images. I don't know how many different things this is! 😅 3? 4? 5? You have a title that says it's "three features" but the issue has 5 different points. Then those 5 points are followed by 2 features that are actually 3 features (but two are duplicates of previous ones), so I'm very confused.

Can you please clean-up your request? :)

**Comment by CanisLupus** [comment]
And like I said in the reply to the review, if you want the es:{searchTerms} to not create a new tab, please use the "Discard on open" checkbox that each search engine has in SSS. It avoids creating a new tab.

![image](https://user-images.githubusercontent.com/532708/103215009-05c18000-490a-11eb-940a-f551eb1ef235.png)

**Comment by drmusti** [comment]
1)possible icon offline import?  

2) Copy to clipboard  and Open as link  icons custom icons addable.

5)https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_iframe 
example sites areas dont work popup....   
![Screenshot_2](https://user-images.githubusercontent.com/33732633/103191448-17912c00-48e6-11eb-8eff-d8c15102f025.png)

https://codepen.io/carwin/details/dxbCi

![Screenshot_3](https://user-images.githubusercontent.com/33732633/103191978-106b1d80-48e8-11eb-9a1e-66a78a464752.png)

**Comment by drmusti** [comment]
i cleaned my request. 
Sincerely.

**Comment by CanisLupus** [comment]
Thank you for the cleanup, it's much better. :)

1. Unfortunately, loading icons from a file doesn't work in the popup due to security policies. This is tracked in issue #135 but I currently don't know if there's a way to solve it.

2. Hmm... While I see why you could want this, it will probably not get implemented in SSS unless another special "engine" like "copy" or "open as link" gets added where this would make sense (in that case we could add support for custom icons on all of them). Right now these special engines are programmed differently from the others internally, and this feature wouldn't be useful for the vast majority of users. :)

3. Thanks for reporting these websites.

    https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_iframe

    Although it looks like an editable text area, I think this site does it weirdly so the text area is actually not editable but there's a floating "textarea" tag of width 1 just so you can write characters one at a time. Even so, that doesn't tell me why the selection doesn't get triggered.
    
    https://codepen.io/carwin/details/dxbCi

    This one is also weird. Perhaps this doesn't work because it is a new HTML document inside an iframe. I'd have to research this more deeply as well.

Cheers!
Daniel

**Comment by CanisLupus** [comment]
I also edited the title of this issue to reflect the issues (except 1, which is in #135).

**Comment by drmusti** [comment]
my icons very great very nice very soft very colorful.
![Screenshot_20](https://user-images.githubusercontent.com/33732633/103302440-bcaa2280-4a14-11eb-8af4-9edd35029d9e.jpg)

**Comment by drmusti** [comment]
new proplem  your translate without formating...  no paragraph  no  <br>  etc...
![Screenshot_2](https://user-images.githubusercontent.com/33732633/103333495-2e1ebb00-4a7f-11eb-9430-603e8e04a450.png)
![Screenshot_3](https://user-images.githubusercontent.com/33732633/103333502-324ad880-4a7f-11eb-9145-04a792e8c447.png)

![Screenshot_4](https://user-images.githubusercontent.com/33732633/103333507-370f8c80-4a7f-11eb-8a1b-56e03f7cb0de.png)

**Comment by CanisLupus** [comment]
> my icons very great very nice very soft very colorful.

Yeah, I imagined that this was why you wanted to customize those icons. :)

> new proplem your translate without formating... no paragraph no etc...

Thanks for reporting this. I've created a new issue (#228) for this, so that we're not dealing with multiple issues here (although there's 2 already ;P).

**Comment by drmusti** [comment]
you are a very kind honest helpful programmer. 
thank you very much big coder.

**Comment by CanisLupus** [comment]
Hahah thank you! ;)

**Comment by drmusti** [comment]
new bug...     error..  
![two proplem](https://user-images.githubusercontent.com/33732633/103540896-89034880-4eab-11eb-90e1-797a5f4146bb.png)

https://yadi.sk/i/aUkSfW5DSYUwWA

wait  page  loaded.
select text.
but you cant a click
button. and you cant 
close popup...
two proplem...

**Comment by CanisLupus** [comment]
Hi @drmusti, I could not reproduce this here. For me it opens, buttons are clickable, and it closes.

Please make sure that you don't have another addon conflicting with SSS (try disabling them and refreshing the page to see if it works). If it doesn't work with everything else disabled, then I'm not sure what it could be.

**Comment by drmusti** [comment]
i found that extention.
Absolute Enable Right Click & Copy
i removet it.
thank you for solution.

**Comment by CanisLupus** [comment]
Ah good, glad it was just that. :)

**Comment by drmusti** [comment]
i too  :))))    
 Absolute Enable Right Click & Copy is not important extention.  basic simple addon...
Sincerely

**Comment by drmusti** [comment]
i found new proplem
after &  lost texts
![Screenshot_1](https://user-images.githubusercontent.com/33732633/103685871-fba12080-4f9e-11eb-93e5-13d4010412fe.png)
![Screenshot_2](https://user-images.githubusercontent.com/33732633/103685882-ff34a780-4f9e-11eb-9190-793d5ed28695.png)
:))))
Sincerely

**Comment by CanisLupus** [comment]
Haha you are good at finding problems with some websites. :)

Thanks for reporting this. It seems that Google Translate doesn't like the conversion of the `&` character if the fragment `#view=home` is present. I had no idea.

Please try this as the engine's Search URL:
`https://translate.google.com/?sl=auto&tl=en&op=translate&text={searchTerms}`

(instead of the old `https://translate.google.com/#view=home&op=translate&sl=auto&tl=en&text={searchTerms}`)

Does this work?

**Comment by CanisLupus** [comment]
(Of course feel free to change the target language from `en` to `tr`)

**Comment by drmusti** [comment]
you are good at finding solutions. great and fast solution :))))  cleverly  congratulations
but translate popup  new line paragraph type proplem still ....
:))))

**Comment by CanisLupus** [comment]
OK, I'll probably change the "translation" engine in the eventual next update to fix this.

True, but the other issues will need a lot more research to fix (if they are even possible to fix). :)

**Comment by drmusti** [comment]
I believe you will be able to solve them all.  because you re best coder big programmer.  
sorry for my bad english.
